cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -219,6 +249,50 @@ private void populateResponse(IThrowableProxy throwableProxy,
     }
   }
 
+  @Nullable
+  private ErrorClassificationResponse findMatchingRule(CloseableIterator<LogEvent> logIter) {
+    while (logIter.hasNext()) {
+      ILoggingEvent logEvent = logIter.next().getLoggingEvent();
+      IThrowableProxy throwableProxy = logEvent.getThrowableProxy();
+      while (throwableProxy != null) {
+        for (ErrorClassificationRule rule : ruleList) {
+          if (throwableProxy.getClassName().equals(rule.getExceptionClassName())
+              && codeMethodMatched(rule.getCodeMethodRegex(),
+              throwableProxy.getStackTraceElementProxyArray())) {
+            LOG.debug("Rule {} matched", rule);
```

Comment:
If you want to log it, add some context. Otherwise it's hard event to say which run is it.

Reply:
removed the log as I think it is not needed and can become chatty.